internal/events/events.go

```diff
@@ -0,0 +1,80 @@
+package events
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/stellar/go/support/log"
+)
+
+var (
+	ErrTopicRequired    = errors.New("message topic is required")
+	ErrKeyRequired      = errors.New("message key is required")
+	ErrTenantIDRequired = errors.New("message tenant ID is required")
+	ErrTypeRequired     = errors.New("message type is required")
+	ErrDataRequired     = errors.New("message data is required")
+)
+
+type Message struct {
+	Topic    string `json:"topic"`
+	Key      string `json:"key"`
+	TenantID string `json:"tenant_id"`
+	Type     string `json:"type"`
+	Data     any    `json:"data"`
+}
+
+func (m Message) String() string {
+	return fmt.Sprintf("Topic: %s - Key: %s - Type: %s - Tenant ID: %s", m.Topic, m.Key, m.Type, m.TenantID)
+}
+
+func (m Message) Validate() error {
+	if m.Topic == "" {
+		return ErrTopicRequired
+	}
+
+	if m.Key == "" {
+		return ErrKeyRequired
+	}
+
+	if m.TenantID == "" {
+		return ErrTenantIDRequired
+	}
+
+	if m.Type == "" {
+		return ErrTypeRequired
+	}
+
+	if m.Data == nil {
+		return ErrDataRequired
+	}
+
+	return nil
+}
+
+type Producer interface {
+	WriteMessages(ctx context.Context, messages ...Message) error
+	Close() error
+}
+
+type Consumer interface {
+	RegisterEventHandler(ctx context.Context, eventHandlers ...EventHandler) error
+	ReadMessage(ctx context.Context) error
+	Close() error
+}
+
+func Consume(ctx context.Context, consumer Consumer) error {
+	log.Ctx(ctx).Info("starting consuming messages...")
+	for {
+		if err := consumer.ReadMessage(ctx); err != nil {
```

Comment:
Usually, you'd want to add ways to break this for loop. For instance, when the context is cancelled or when the OS sends a SIGINT.

We're currently doing that in the TSS, so you can replicate that:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/36ed4b5f9b88968ca12bf5dbd1ab16883541aff0/internal/transactionsubmission/manager.go#L216-L222

Reply:
@CaioTeixeira95 This is a good suggestion I think. We should add the cancellation.

Reply:
changed: [e3e7307](https://github.com/stellar/stellar-disbursement-platform-backend/pull/119/commits/e3e7307805afd96364e340a49ba58919bec502c4)